src/components/NewClusterModal.js

```diff
@@ -110,41 +107,38 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
   constructor(props) {
     super(props)
     const { currentCluster } = props
-    const currentConfig = currentCluster ? currentCluster.machineConfig : profiles[0].machineConfig
-    const matchingProfile = _.find(
-      ({ machineConfig }) => machineConfigsEqual(machineConfig, currentConfig),
-      profiles
-    )
+    const { cloudService, ...currentConfig } = normalizeRuntimeConfig(currentCluster?.runtimeConfig || profiles[0].runtimeConfig)
+    const { masterDiskSize, masterMachineType, numberOfWorkers } = currentConfig // want these to be put into state below, unlike cloudService
+    const matchingProfile = masterDiskSize === normalizeRuntimeConfig({}).masterDiskSize && _.find({ runtimeConfig: { masterMachineType } }, profiles)
```

Comment:
We probably shouldn't assume that the preset profiles always use the default disk size.
```suggestion
    const matchingProfile = _.find(({ runtimeConfig }) => _.isMatch({ masterMachineType, masterDiskSize }, normalizeRuntimeConfig(runtimeConfig)), profiles)
```